frontend/src/entities/configuration/dynamic-configuration.tsx

```diff
@@ -136,15 +216,211 @@ export function DynamicConfiguration () {
                     {<pre>TODO</pre>}
                 </ExpandableSection>
                 <ExpandableSection headerText='EMR Config' variant='default' defaultExpanded>
-                    {<pre>TODO</pre>}
-                    <Input
-                        data-cy='test-input'
-                        value={state.form.configuration.EMRConfig.clusterSizes[0].name}
-                        onChange={(event) => {
-                            setFields({ 'configuration.EMRConfig.clusterSizes[0].name': event.detail.value });
-                        }}
-                        onBlur={() => touchFields(['configuration.EMRConfig.clusterSizes[0].name'])}
-                    />
+                    <ExpandableSection 
+                        headerText='Applications'
+                        variant='default' 
+                        headingTagOverride='h3' 
+                        headerDescription='A list of applications for Amazon EMR to install and configure when launching the cluster.'
+                    >
+                        <Multiselect
+                            selectedOptions={selectedApplicationOptions}
+                            onChange={({ detail }) =>
+                                addApplication(detail)
+                            }
+                            options={applicationOptions}
+                            placeholder='Select applications'
+                        />
+                    </ExpandableSection>
+                    <ExpandableSection 
+                        headerText='Cluster Types'
+                        variant='default' 
+                        headingTagOverride='h3' 
+                        headerDescription='The cluster types options that users can select from when creating a new Amazon EMR Cluster.'
+                    >
+                        <ClusterTypeConfiguration
+                            item={
+                                state.form as (IAppConfiguration)
```

Comment:
Above if you supply the type you won't have to cast this. If you use `useValidationReducer<IAppConfiguration>(...)` it should know `state.form` is that type. Actually since applicationConfiguration is already that type you can probably just do `form: applicationConfiguration` right?

Reply:
Looks like I can just remove the `as (IAppConfiguration)` altogether and it's fine.